topics/ui/principles/empty_state.md

```diff
@@ -10,144 +10,261 @@
+<img src="empty_state_master_detail_layout.png" alt="Empty master-detail layout" width="706"/>
 
-<format color="Green" style="bold">Correct</format>
 
-![](todo-filters-correct.png){width=456 style=block}
-*Place an inline help text under the table.*
 
-### Writing guidelines
+## Writing guidelines
 
-See [Punctuation](punctuation.md) and [Capitalization](capitalization.md).
+### Punctuation
 
-Make the reason text short and descriptive. See [Writing short and clear](writing_short.md).
+Separate sentences in the empty state with periods, but do not put a period after the action link.
+Use the ellipses at the end of the action link if this link opens a dialog or a popup which requires some input from the user.
+<table style="none" border="false" column-width="fixed">
+  <tr>
+    <td>
+      <format color="369650" style="bold">Correct</format>
+      <img src="empty_state_punctuation_correct.png" alt="Correct punctuation in the empty state"/>
+    </td>
+    <td>
+      <format color="E55765" style="bold">Incorrect</format>
+      <img src="empty_state_punctuation_incorrect.png" alt="Incorrect punctuation in the empty state"/>
+    </td>
+  </tr>
+</table>
 
-In actions, avoid words that describe physical actions like _press_ or _click_ — they are obvious from how the link works.
+### Capitalization
+
+Use sentence capitalization in the empty state texts and links.
+<table style="none" border="false" column-width="fixed">
+  <tr>
+    <td>
+      <format color="369650" style="bold">Correct</format>
+      <img src="empty_state_capitalization_correct.png" alt="Correct capitalization in the empty state"/>
+    </td>
+    <td>
+      <format color="E55765" style="bold">Incorrect</format>
+      <img src="empty_state_capitalization_incorrect.png" alt="Incorrect capitalization in the empty state"/>
+    </td>
+  </tr>
+</table>
 
-Avoid saying _add new_. Just use _add_ because all that is added is new in the context of an empty UI area.
+### Avoid excessive verbs
 
-## Master-detail layout
+- In actions, avoid words that describe physical actions like _press_ or _click_ — they are obvious from how the link works.
 
-Provide only the action part for the detail area in a master-detail layout. The detail area is filled when an item is selected in the master part. This behavior is obvious and does not need to be explained.
+- Avoid saying _add new_. Just use _add_ because all that is added is new in the context of an empty UI area.
 
-The default pattern for the action is "_Select_ [entity] _to configure_".
+<table style="none" border="false" column-width="fixed">
+  <tr>
+    <td>
+      <format color="369650" style="bold">Correct</format>
+      <img src="empty_state_verbs_correct.png" alt="Correct usage of verbs in the empty state"/>
+    </td>
+    <td>
+      <format color="E55765" style="bold">Incorrect</format>
+      <img src="empty_state_verbs_incorrect.png" alt="Incorrect usage of verbs in the empty state"/>
+    </td>
+  </tr>
+</table>
 
```

Comment:
Possibly return info about non-breaking spaces? https://plugins.jetbrains.com/docs/intellij/empty-state.html#sizes-and-placement

It's a small detail but makes the typography a bit more polished